cpp/bench/common/benchmark.hpp

```diff
@@ -122,6 +122,13 @@ class fixture {
   virtual void run_benchmark(::benchmark::State& state) = 0;
   virtual void generate_metrics(::benchmark::State& state) {}
 
+ protected:
+  /** The helper that writes zeroes to some buffer in GPU memory to flush the L2 cache.  */
+  void flush_L2_cache()
+  {
+    RAFT_CUDA_TRY(cudaMemsetAsync(scratch_buf_.data(), 0, scratch_buf_.size(), stream));
```

Comment:
Just to be sure, it is better to set the size of this buffer to be 3x that of the size of the L2-cache.

Reply:
Sure, why not :)